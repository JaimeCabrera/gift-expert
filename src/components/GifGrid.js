import React from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

   // para renombrar en la desetructuracion name:newname
   const {data:images, loading} = useFetchGifs(category);
   // const [images, setImages] = useState([]);

   // useEffect(() => {
   //    getGifs(category).then(gifs => setImages(gifs));
   // }, [category])

   // recibimos la categoria y buscamos los gifs de esa categoria
   
   return (
      <>
         <h3 className="animate__animated animate__fadeIn">{category}</h3>
         {loading && <p className="animate__animated animate__flash">Cargando...</p>}
         <div className="card-grid">
            {
               images.map(img => {
                  // return <li key={id}>{title}</li>
                  return <GifGridItem
                     key={img.id}
                     {...img}
                  />
               })
            }

         </div>
      </>
   )
}

GifGrid.propTypes = {
   category:PropTypes.string
}

export default GifGrid
