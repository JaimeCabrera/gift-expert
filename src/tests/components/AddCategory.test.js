import React from 'react'
import { shallow } from "enzyme";
import { default as AddCategory } from "../../components/AddCategory";

describe('Pruebas en el componente <AddCategory/>', () => {
   
   const setCategories = jest.fn();//para referenciar una funcion con jest
   let wrapper = shallow(<AddCategory setCategories={setCategories}/>);
   
   // ciclo de vida de la funciones
   beforeEach(() => {
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory setCategories={setCategories} />);

   })
   
   test('debe mostrar correctamente el componente', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('Debe de cambiar la caja de texto', () => {
      const input = wrapper.find('input');
      const value = "hola mundo";
      input.simulate('change',{target:{value}});
   });

   test('no debe de postear la informacion con submit', () => {
      wrapper.find('form').simulate('submit', { preventDefault() { } });
      expect(setCategories).not.toBeCalled();
   });

   test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
      const value = "futurama";
      // 1. simular el inputChange
      wrapper.find('input').simulate('change', { target: { value } });
      // 2. simular el submit del formulario
      wrapper.find('form').simulate('submit', { preventDefault() { } });
      // 3. setCategories 1 vez
      expect(setCategories).toHaveBeenCalled();
      
      // expect(setCategories).toHaveBeenCalledTimes(2);

      // 4. el value del input debe estar vacio
      expect(wrapper.find('input').prop('value')).toBe('');

   })
   
   
   
   
   

})
