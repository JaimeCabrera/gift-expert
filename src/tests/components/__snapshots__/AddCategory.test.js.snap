// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory/> debe mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
